.buildkite/pipelines/pull_request/base.yml

```diff
@@ -86,6 +86,18 @@ steps:
         - exit_status: '-1'
           limit: 3
 
+  - command: .buildkite/scripts/steps/checks/prompt_changes_detector.sh
```

Comment:
We have a step for change detection running already, would this fit there?

`.buildkite/scripts/pipelines/pull_request/pipeline.ts` checks for changed files and uploads the pipeline step.

Alternatively, quick_checks can always run .buildkite/scripts/steps/checks/prompt_changes_detector.sh.   

I'm mostly adding this because there's a few minutes of overhead for each CI agent while the repository is checked out and bootstrapped.  Not a huge impact but it adds up over time.